src/position/position.js

```diff
@@ -39,6 +41,66 @@ angular.module('ui.bootstrap.position', [])
       return offsetParent || docDomEl;
     };
 
+    /**
+     * returns desired positional information for element
+     * @param position - object
+     * @param defaultPosition - object
+     */
+    function constructPlacement(position, positionStr, offset) {
+      var newPosition = {
+        width: position.width,
+        height: position.height
+      };
+
+      if (angular.isString(positionStr)) {
+        // When absolutely positioning in relative parent,
+        // 'top' means pushing from bottom above target element
+        if (offset === 'position') {
+          positionStr = positionStr.replace(VERTICAL_REGEX, function(match, first) {
+            return first ? 'bottom' : 'top';
+          });
+        }
+        /*
+         * POSITION_REGEX
+         * 1: top/bottom & optional second position
+         * 2: top/bottom
+         * 3: spaces & left/right
+         * 4: left/right (with top/bottom)
+         * 5: left/right (without top/bottom)
+         */
+        var match = positionStr.match(POSITION_REGEX);
+        if (match) {
+          if (match[5]) {
+            if (match[5] === 'right') {
+              newPosition.right = 0;
+            } else {
+              newPosition.left = 0;
+            }
+          } else {
+            if (match[2] === 'bottom') {
+              newPosition.bottom = newPosition.height - position.top;
+            } else {
+              newPosition.top = offset === 'offset' ? 0 : position.top;
+            }
+
+            if (match[4] && match[4] === 'right') {
+              newPosition.right = 0;
+            } else {
+              newPosition.left = 0;
+            }
+          }
+        } else {
+          newPosition.top = offset === 'offset' ? 0 : position.top;
+          newPosition.left = position.left;
+        }
+      } else {
+        newPosition.top = offset === 'offset' ? 0 : position.top;
+        newPosition.left = position.left;
```

Comment:
If it's the default case, perhaps it be handled via a `isInvalid` flag or something, and a check below, since the code is identical, might make sense to put it together.